internal/ipmi/uuid.go

```diff
@@ -90,26 +90,18 @@ func parseUUIDLine(l string) string {
 	return strings.ToLower(uuidRegexCompiled.FindString(l))
 }
 
-func (u UUIDCache) loadUUID(ip, user, password, sum string) (string, error) {
-	args := []string{"--no_banner", "--no_progress", "--journal_level", "0", "-i", ip, "-u", user, "-p", password, "-c", "GetDmiInfo"}
-	cmd := exec.Command(sum, args...)
-	out, err := cmd.StdoutPipe()
+func (u UUIDCache) loadUUID(ip string, port int, user, password string) (string, error) {
+	ob, err := detect.ConnectOutBand(ip, port, user, password)
 	if err != nil {
-		return "", fmt.Errorf("could not initiate sum command to get dmi data from ip:%s, err: %v", ip, err)
+		return "", errors.Wrapf(err, "could not open out-band connection to ip:%s, port:%d, user: %s", ip, port, user)
 	}
-	err = cmd.Start()
+
+	info, err := ob.DmiInfo()
 	if err != nil {
-		return "", fmt.Errorf("could not start sum command to get dmi data from ip:%s, err: %v", ip, err)
+		return "", errors.Wrapf(err, "failed to get dmi data from ip:%s", ip)
 	}
-	go func() {
-		err = cmd.Wait()
-		if err != nil {
-			u.log.Errorw("loadUUID", "ip", ip, "wait error", err)
-		}
-	}()
-	s := bufio.NewScanner(out)
-	for s.Scan() {
-		l := s.Text()
+
+	for _, l := range info {
```

Comment:
I thought these informations where also available via the redfish api, and then we do not need to make a supermicro proprietary sum call to get them.

Reply:
Then we need the redfish client also for supermicro. I'll check this out

Reply:
I'll also check if we can get the machine UUID for supermicro via redfish as well.
If both works, go-hal does not need to use the sum tool anymore.